### PR TITLE
Boost Plugin: Add a dependency on the automattic/jetpack-composer-plugin Composer package

### PR DESCRIPTION
**Pull request by kbrown9**
#### Changes proposed in this Pull Request:
* Add a dependency on the automattic/jetpack-composer-plugin package. This package will install `jetpack-library` packages in the `jetpack_vendor` directory, which will allow the packages' strings to be extracted for translation.

#### Jetpack product discussion
* n/a

#### Does this pull request change what data or activity we track or use?
* no

##### Check the Development Build in the Jetpack Docker development environment.
1. Checkout this branch.
2. Delete the existing vendor folder (`projects/plugins/boost/vendor`).
3. In the `projects/plugins/boostp` directory, run `composer update automattic/jetpack-composer-plugin`.
4. Verify that 15 packages have been installed in the `projects/plugins/boost/jetpack_vendor/automattic` directory.
5. Verify that 4 packages (`jetpack-autoloader`, `jetpack_changelogger`, `jetpack-composer-plugin`, and `wordbless`) have been installed in the `projects/plugins/boost/vendor/automattic` directory.

##### Check the Production build
1. Download the production build of this branch from the Jetpack beta build server: https://betadownload.jetpack.me/data/jetpack-boost/update_boost_composer_package_i18/jetpack-boost-dev.zip
2. Verify that 15 packages have been installed in the `jetpack_vendor/automattic` directory.
3. Verify that no packages have been installed in the `vendor/automattic` directory.

### CODE REVIEW

**Review by anomiex — APPROVED** [review]
LGTM. Verified that the built artifact doesn't have any unexpected changes.

**Comment by kbrown9** [comment]
This needs another review because I had to update the `composer.lock` file. I've enabled auto-merge so this will merge as soon as it has an approval.

**Review by anomiex — APPROVED** [review]

**Pull request merged**
